### DISCUSSION TITLE
Managing Independent Games Within an Android App : Handling Folder Structure and Execution Paths

### DISCUSSION THREAD

**Discussion by mzelecom** [godotengine/godot-proposals#11346] [Platforms]
I want to embed small games inside an Android app. The issue I’m facing right now is that the project files need to be in the assets folder.

If there’s a feature that allows us to specify the execution folder to the engine at the start, we can manage this situation.

Additionally, the game files might be compressed and hosted online, and with this feature, we could download the files separately and set the execution path in the app so the game loads from the desired folder.

If I want to load separate games, it’s not possible right now, and I want to organize them into separate folders, keeping them independent of each other.

Thank you.

**Comment by syntaxerror247** [comment]
I have never really tried something like this. I assume you are embedding Godot inside an Android app. Right?

**Reply by mzelecom** [reply]
yes.
i used godot-4.3.0.stable.aar to load game in my aplication it was ok .
but i need download game zip file and uncompress in internal folder to use.
godot defult read game from assets folder.i need change it.

**Comment by syntaxerror247** [comment]
@mzelecom I think you need to use .pck or .zip to load your games.

1. Export your Godot project as .pck file.
2. Use `ProjectSettings.load_resource_pack("path to pck file")`

See: https://docs.godotengine.org/en/stable/tutorials/export/exporting_pcks.html

**Reply by mzelecom** [reply]
in android library i cant use pck .
i just start activity and godot run game.
this is my code

`package com.godot.game;
import org.godotengine.godot.GodotActivity;
import android.os.Bundle;
import androidx.core.splashscreen.SplashScreen;


public class GodotApp extends GodotActivity {
	@Override
	public void onCreate(Bundle savedInstanceState) {
		SplashScreen.installSplashScreen(this);
		super.onCreate(savedInstanceState);
	}
}`

**Reply by syntaxerror247** [reply]
But you can use it inside your Godot project that is in assets folder by default.

Keep the default scene to scan and load games. Also add a option to load new games using .pck

**Reply by mzelecom** [reply]
i cant chnage defult assets?
just change file_access_android.cpp file to change assets?

**Reply by syntaxerror247** [reply]
No, I don't think so.

**Reply by syntaxerror247** [reply]
Why can't you load using .pck for .zip? This is created exactly for adding additional content to deployed games.
